src/viam/sdk/components/board/client.cpp

```diff
@@ -23,151 +24,71 @@ BoardClient::BoardClient(std::string name, std::shared_ptr<grpc::Channel> channe
+                  request.set_pin(pin);
+                  request.set_duty_cycle_pct(duty_cycle_pct);
+              })
+        .invoke();
 }
 
 uint64_t BoardClient::get_pwm_frequency(const std::string& pin, const AttributeMap& extra) {
-    viam::component::board::v1::PWMFrequencyRequest request;
-    viam::component::board::v1::PWMFrequencyResponse response;
-
-    grpc::ClientContext ctx;
-    set_client_ctx_authority(ctx);
-
-    *request.mutable_name() = this->name();
-    *request.mutable_extra() = map_to_struct(extra);
-    request.set_pin(pin);
-
-    const grpc::Status status = stub_->PWMFrequency(&ctx, request, &response);
-    if (!status.ok()) {
-        throw std::runtime_error(status.error_message());
-    }
-    return response.frequency_hz();
+    return make_client_helper(this, *stub_, &StubType::PWMFrequency)
+        .with(extra, [&](auto& request) { request.set_pin(pin); })
+        .invoke([](auto& response) { return response.frequency_hz(); });
 }
 
 void BoardClient::set_pwm_frequency(const std::string& pin,
                                     uint64_t frequency_hz,
                                     const AttributeMap& extra) {
-    viam::component::board::v1::SetPWMFrequencyRequest request;
-    viam::component::board::v1::SetPWMFrequencyResponse response;
-
-    grpc::ClientContext ctx;
-    set_client_ctx_authority(ctx);
-
-    *request.mutable_name() = this->name();
-    *request.mutable_extra() = map_to_struct(extra);
-    request.set_pin(pin);
-    request.set_frequency_hz(frequency_hz);
-
-    const grpc::Status status = stub_->SetPWMFrequency(&ctx, request, &response);
-    if (!status.ok()) {
-        throw std::runtime_error(status.error_message());
-    }
+    return make_client_helper(this, *stub_, &StubType::SetPWMFrequency)
+        .with(extra,
+              [&](auto& request) {
+                  request.set_pin(pin);
+                  request.set_frequency_hz(frequency_hz);
+              })
+        .invoke();
 }
 
 AttributeMap BoardClient::do_command(const AttributeMap& command) {
-    viam::common::v1::DoCommandRequest request;
-    viam::common::v1::DoCommandResponse response;
-
-    grpc::ClientContext ctx;
-    set_client_ctx_authority(ctx);
-
-    const google::protobuf::Struct proto_command = map_to_struct(command);
-    *request.mutable_command() = proto_command;
-    *request.mutable_name() = this->name();
-
-    const grpc::Status status = stub_->DoCommand(&ctx, request, &response);
-    if (!status.ok()) {
-        throw std::runtime_error(status.error_message());
-    }
-    return struct_to_map(response.result());
+    return make_client_helper(this, *stub_, &StubType::DoCommand)
+        .with([&](auto& request) { *request.mutable_command() = map_to_struct(command); })
+        .invoke([](auto& response) { return struct_to_map(response.result()); });
 }
 
+// TODO(RSDK-6048) update `client_wrapper` to allow for requests without a `mutable_name()` method,
+// then wrap here.
```

Comment:
My editor didn't catch it but this problem also prevents us from using `service_wrapper` for this method in the server code, will be backing that out as well.

Reply:
Makes sense! I'll add a comment to the ticket to that effect.